jupyterlab_automl/jupyterlab_automl/service.py

```diff
@@ -1,284 +1,243 @@
+  """Provides an authenticated AutoML Client and project info"""
+
+  _instance = None
+
+  def __init__(self):
+    client_options = {"api_endpoint": API_ENDPOINT}
+    self._dataset_client = aiplatform_v1alpha1.DatasetServiceClient(
+        client_options=client_options)
+    self._model_client = aiplatform_v1alpha1.ModelServiceClient(
+        client_options=client_options)
+    self._pipeline_client = aiplatform_v1alpha1.PipelineServiceClient(
+        client_options=client_options)
+    self._gcs_client = storage.Client()
+    self._project = AuthProvider.get().project
+    self._parent = "projects/{}/locations/us-central1".format(self._project)
+    self._gcs_bucket = "jupyterlab-ucaip-data-{}".format(
+        hashlib.md5(self._project.encode('utf-8')).hexdigest())
+
+  @property
+  def dataset_client(self):
+    return self._dataset_client
+
+  @property
+  def model_client(self):
+    return self._model_client
+
+  @classmethod
+  def get(cls):
+    if not cls._instance:
+      cls._instance = AutoMLService()
+    return cls._instance
+
+  def get_models(self):
+    models = []
+    for model in self._model_client.list_models(parent=self._parent).models:
+      model_type = 'OTHER'
+      if 'image' in model.metadata_schema_uri:
+        model_type = 'IMAGE'
+      elif 'tables' in model.metadata_schema_uri:
+        model_type = 'TABLE'
+      models.append({
+            "id": model.name,
+            "displayName": model.display_name,
+            "pipelineId": model.training_pipeline,
+            "createTime": get_milli_time(model.create_time),
+            "updateTime": get_milli_time(model.update_time),
+            "etag": model.etag,
+            "modelType": model_type
+        })
+    return models
+
+  def _get_feature_importance(self, model_explanation):
+    features = model_explanation["meanAttributions"][0]["featureAttributions"].items()
+    return [
+        {
+            "name": key,
+            "Percentage": round(val * 100, 3),
+        } for key, val in features
+    ]
+
+  def _get_confusion_matrix(self, confusion_matrix):
+    rows = confusion_matrix["rows"]
+    titles = []
+    for column in confusion_matrix["annotationSpecs"]:
+      titles.append(column["displayName"])
+    rows.insert(0, titles)
+    return rows
+
+  def _get_confidence_metrics(self, confidence_metrics):
+    labels = ["confidenceThreshold", "f1Score", "f1ScoreAt1", "precision",
+              "precisionAt1", "recall", "recallAt1", "trueNegativeCount",
+              "truePositiveCount", "falseNegativeCount", "falsePositiveCount",
+              "falsePositiveRate", "falsePositiveRateAt1"]
+    metrics = []
+    for confidence_metric in confidence_metrics:
+      metric = {}
+      for label in labels:
+        if label == "confidenceThreshold":
+          value = confidence_metric.get(label, 0.0) * 100
         else:
-            return []
+          value = confidence_metric.get(label, "NaN")
+        metric[label] = value
```

Comment:
```        
if label == "confidenceThreshold":
  value = confidence_metric.get(label, 0.0) * 100
else:
  value = confidence_metric.get(label, "NaN")
metric[label] = value
```
The code looks like this - it looks like we're setting the value variable on each iteration with a different default based on the label value. Let me know if this makes more sense

Reply:
I think the reasoning here is because the confidenceThreshold needs to have a default as a number instead of a string because of the logic in the TypeScript

Reply:
I see - so given that we're doing the same thing for all cases except "confidenceThreshold", this pattern might be more readable (and a slight bit more efficient):

```Python
for label in labels:
  metric[label] = confidence_metric.get(label, "NaN")
metric["confidenceThreshold"] = confidence_metric.get(label, 0.0) * 100
```

Reply:
one problem with this approach is that the confidenceThreshold field might not always exist on the confidence_metric - for example on regression models

Reply:
I think this function won't even be called for regression models because of ```    if "confidenceMetrics" in metrics:
      model_eval["confidenceMetrics"] = self._get_confidence_metrics(metrics["confidenceMetrics"])```

Reply:
oh nevermind that makes sense!

Reply:
Also, ```labels``` is a predefined list in the method, so either way I think they're logically equivalent.